src/codegen/codegen_c.cc

```diff
@@ -568,6 +568,7 @@ void CodeGenC::VisitExpr_(const Load* op, std::ostream& os) {  // NOLINT(*)
       os << ref;
     } else {
       // load seperately.
+      int vec_scope = BeginScope();
       std::string svalue = GetUniqueName("_");
```

Comment:
add a comment:

The assignment introduces side-effect, and the resulting value cannot be reused across multiple expression, thus a new scope is needed